network-store-server/src/main/java/com/powsybl/network/store/server/NetworkStoreRepository.java

```diff
@@ -312,106 +134,149 @@ void prepareStatements() {
+                                                           resultSet.getString(2)));
+                    }
+                    return networksInfos;
+                }
             }
-            return networksInfos;
+        } catch (SQLException e) {
+            throw new UncheckedSqlException(e);
         }
     }
 
     public List<VariantInfos> getVariantsInfos(UUID networkUuid) {
-        try (ResultSet resultSet = session.execute(selectFrom(NETWORK).columns(VARIANT_ID, VARIANT_NUM)
-                .whereColumn(UUID_STR).isEqualTo(literal(networkUuid)).build())) {
-            List<VariantInfos> variantsInfos = new ArrayList<>();
-            for (Row row : resultSet) {
-                variantsInfos.add(new VariantInfos(row.getString(0), row.getInt(1)));
+        try (var connection = dataSource.getConnection()) {
+            try (var preparedStmt = connection.prepareStatement(QueryCatalog.buildGetVariantsInfos())) {
+                preparedStmt.setObject(1, networkUuid);
+                try (ResultSet resultSet = preparedStmt.executeQuery()) {
+                    List<VariantInfos> variantsInfos = new ArrayList<>();
+                    while (resultSet.next()) {
+                        variantsInfos.add(new VariantInfos(resultSet.getString(1), resultSet.getInt(2)));
+                    }
+                    return variantsInfos;
+                }
             }
-            return variantsInfos;
+        } catch (SQLException e) {
+            throw new UncheckedSqlException(e);
         }
     }
 
     public Optional<Resource<NetworkAttributes>> getNetwork(UUID uuid, int variantNum) {
-        Map<String, Mapping> mappingNetworks = mappings.getNetworkMappings().getColumnMapping();
-        Set<String> columnMappings = new HashSet<>(mappingNetworks.keySet());
-        columnMappings.add(ID_STR);
-        try (ResultSet resultSet = session.execute(selectFrom(NETWORK)
-            .columns(columnMappings.toArray(new String[0]))
-            .whereColumn(UUID_STR).isEqualTo(literal(uuid))
-            .whereColumn(VARIANT_NUM).isEqualTo(literal(variantNum))
-            .build())) {
-            Row one = resultSet.one();
-            if (one != null) {
-                NetworkAttributes networkAttributes = new NetworkAttributes();
-                mappingNetworks.forEach((key, value) -> value.set(networkAttributes, one.get(key, value.getClassR())));
-                return Optional.of(Resource.networkBuilder()
-                    .id(one.get(ID_STR, String.class))
-                    .variantNum(variantNum)
-                    .attributes(networkAttributes)
-                    .build());
+        var networkMapping = mappings.getNetworkMappings();
+        try (var connection = dataSource.getConnection()) {
+            var preparedStmt = connection.prepareStatement(QueryCatalog.buildGetNetworkQuery(networkMapping.getColumnMapping().keySet()));
```

Comment:
try with resource for the preparedstatement ?

Reply:
Done